pkg/policy/l4.go

```diff
@@ -724,8 +727,34 @@ func createL4Filter(policyCtx PolicyContext, peerEndpoints api.EndpointSelectorS
 			}
 		}
 
+		// Override the parser type to CRD is applicable.
+		forceRedirect := false
+		if pr.Listener != nil {
+			l4.L7Parser = ParserTypeCRD
+			ns := policyCtx.GetNamespace()
+			resource := pr.Listener.EnvoyConfig
+			switch resource.Kind {
+			case "CiliumEnvoyConfig":
+				if ns == "" {
+					// Cluster-scoped CCNP tries to use namespaced
+					// CiliumEnvoyConfig
+					//
+					// TODO: Catch this in rule validation once we have a
+					// validation context in there so that we can differentiate
+					// between CNP and CCNP at validation time.
```

Comment:
Unfortunately, this may not be possible to do at CRD-level validation, because the CCNP type is embedded inside the CNP type. This means that the validation is simply copied over to CCNP. This has it's pros and cons, because 99% of cases, a CCNP should be validated like a CNP as they're pretty much the same thing, except when they are not. 

It's also not as easy as just not embedding the type because there are a lot of places in the code which assumes this structure, last I looked (>1y).